client/landing/stepper/declarative-flow/internals/steps-repository/woo-transfer/index.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-console */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { transferStates } from 'calypso/state/atomic/transfers/constants';
+import { ONBOARD_STORE, SITE_STORE } from '../../../../stores';
+import type { Step } from '../../types';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+const WooTransfer: Step = function WooTransfer( { navigation } ) {
+	const { submit } = navigation;
+	const { setPendingAction, setProgress } = useDispatch( ONBOARD_STORE );
+	const {
+		requestAtomicSoftwareStatus,
+		requestLatestAtomicTransfer,
+		initiateAtomicTransfer,
+	} = useDispatch( SITE_STORE );
+	const site = useSite();
+
+	const siteId = site?.ID;
+
+	const {
+		getSiteLatestAtomicTransfer,
+		getSiteLatestAtomicTransferError,
+		getAtomicSoftwareStatus,
+		// getAtomicSoftwareError,
+	} = useSelect( ( select ) => select( SITE_STORE ) );
+
+	useEffect( () => {
+		if ( ! siteId ) return;
+
+		setPendingAction( async () => {
+			const startTime = new Date().getTime();
+			const totalTimeout = 1000 * 180;
+			const maxFinishTime = startTime + totalTimeout;
+
+			// Initiate transfer
+			await initiateAtomicTransfer( siteId, 'woo-on-plans' );
+
+			// Poll for transfer status
+			let stopPollingTransfer = false;
+
+			while ( ! stopPollingTransfer ) {
+				await wait( 3000 );
+				await requestLatestAtomicTransfer( siteId );
```

Comment:
Why is `await` needed here?

Reply:
It's not! 
With that said, it returns a Promise. I assumed it's resolved after the request is over and the state is updated.
If this is the case, the next line would fetch the latest status (and we wouldn't waste time `wait`ing for something that we already have). 
If it doesn't really wait for the state to be updated before resolving, there's no harm.
But I _do_ intend to test this assumption 😄